.github/workflows/ci.yml

```diff
@@ -49,10 +49,10 @@ jobs:
       - name: run pre-commit
         if: ${{ matrix.python-version == env.LATEST_PY_VERSION }}
         run: |
-          uv run pre-commit run --all-files
+          uv run pre-commit run --all-files --verbose --show-diff-on-failure
 
       - name: Run tests
-        run: uv run pytest
+        run: uv run pytest -n auto
```

Comment:
Added pytest-xdist mostly for more performant tests locally. Does not do much on a 2 core gh runner.